src/sql/src/plan/statement/ddl.rs

```diff
@@ -1407,7 +1411,217 @@ pub fn plan_create_views(
+                            publication: _,
+                            details,
+                            ..
+                        }),
+                    ..
+                } => {
+                    // If the user specified targets, validate they are all in the PostgresSourceDetails
+                    // otherwise create them from the contents of PostgresSourceDetails
+                    let targets = match targets {
+                        Some(targets) => {
+                            let known: HashSet<String> =
+                                HashSet::from_iter(details.tables.iter().flat_map(|t| {
+                                    // It is easier to put both schema_name.table_name and table_name into the hashset than to peel off the namespace from a string
+                                    vec![
+                                        format!("{}", UnresolvedObjectName::qualified(&[&t.name])),
+                                        format!(
+                                            "{}",
+                                            UnresolvedObjectName::qualified(&[
+                                                &t.namespace,
+                                                &t.name
+                                            ])
+                                        ),
+                                    ]
+                                }));
+                            let wanted: HashSet<String> =
+                                HashSet::from_iter(targets.iter().map(|t| t.name.to_string()));
+                            let diff = wanted.difference(&known);
+                            if diff.clone().count() > 0 {
+                                error!("During create views user specified tables not found: {:?}  Known tables: {:?}", diff, known);
+                                return Err(anyhow!(
+                                    "Table(s) {} not found in source, try recreating materialized source.",
+                                    diff.map(|d| d.to_owned())
+                                        .collect::<Vec<String>>()
+                                        .join(", ")
+                                ));
+                            }
+                            targets
+                        }
+                        None => details
+                            .tables
+                            .iter()
+                            .map(|t| {
+                                let name =
+                                    UnresolvedObjectName::qualified(&[&t.namespace, &t.name]);
+                                CreateViewsSourceTarget {
+                                    name: name.clone(),
+                                    alias: Some(name),
+                                }
+                            })
+                            .collect(),
+                    };
+
+                    let mut viewdefs = Vec::with_capacity(details.tables.len());
```

Comment:
shouldn't this be `targets.len()`?

Reply:
Good call, that may be smaller than the full set of tables in details.